kraken/node_actions/openstack_node_scenarios.py

```diff
@@ -0,0 +1,121 @@
+
+    # Get instance status
+    def get_instance_status(self, node, expected_status, timeout):
+        i = 0
+        sleeper = 1
+        while i <= timeout:
+            instStatus =  runcommand.invoke("openstack server show %s | tr -d ' ' | grep '^|status' | cut -d '|' -f3 | tr -d '\n'" % (node))
+            logging.info("instance status is %s" % (instStatus))
+            logging.info("expected status is %s" % (expected_status))
+            if (instStatus.strip() == expected_status):
+                logging.info("instance status has reached desired status %s" % (instStatus))
+                return True
+            time.sleep(sleeper)
+            i += sleeper
+    
+    # Get the openstack instance name
+    def get_openstack_nodename(self,os_node_ip):
+        action_output = runcommand.invoke("openstack server list | grep %s | cut -f3 -d '|' | tr -d '\040\011\012\015\174'" % (os_node_ip))
+        logging.info("OPENSTACKCLOUD node name is %s" % (action_output))
+        return action_output
+        
+class openstack_node_scenarios(abstract_node_scenarios):
+    def __init__(self):
+        self.openstackcloud = OPENSTACKCLOUD()
+
+    # Node scenario to start the node
+    def node_start_scenario(self, instance_kill_count, node, timeout):
+        for _ in range(instance_kill_count):
+            try:
+                logging.info("Starting node_start_scenario injection")
+                logging.info("Starting the node %s" % (node))
+                openstack_node_ip = nodeaction.get_node_ip(node)
+                openstack_node_name = self.openstackcloud.get_openstack_nodename(openstack_node_ip)
+                self.openstackcloud.start_instances(openstack_node_name)
+                self.openstackcloud.wait_until_running(openstack_node_name)
+                nodeaction.wait_for_ready_status(node, timeout)
+                logging.info("Node with instance ID: %s is in running state" % (node))
+                logging.info("node_start_scenario has been successfully injected!")
+            except Exception as e:
+                logging.error("Failed to start node instance. Encountered following "
+                              "exception: %s. Test Failed" % (e))
+                logging.error("node_start_scenario injection failed!")
+                sys.exit(1)
+
+    # Node scenario to stop the node
+    def node_stop_scenario(self, instance_kill_count, node, timeout):
+        for _ in range(instance_kill_count):
+            try:
+                logging.info("Starting node_stop_scenario injection")
+                logging.info("Stopping the node %s " % (node))
+                openstack_node_ip = nodeaction.get_node_ip(node)
+                openstack_node_name = self.openstackcloud.get_openstack_nodename(openstack_node_ip)
+                self.openstackcloud.stop_instances(openstack_node_name)
+                self.openstackcloud.wait_until_running(openstack_node_name)
```

Comment:
Should be `wait_until_stopped` instead of running

Reply:
Good catch.  I will change it.